src/mono/wasm/runtime/invoke-js.ts

```diff
@@ -328,13 +329,16 @@ export async function dynamic_import(module_name: string, module_url: string): P
         promise = import(/* webpackIgnore: true */module_url);
         importedModulesPromises.set(module_name, promise);
     }
-    const module = await promise;
-    if (newPromise) {
-        importedModules.set(module_name, module);
-        if (runtimeHelpers.diagnosticTracing)
-            console.debug(`MONO_WASM: imported ES6 module '${module_name}' from '${module_url}'`);
-    }
-    return module;
+
+    return wrap_as_cancelable_promise(async () => {
```

Comment:
Are there any other cases that need a similar wrapper?

Reply:
Great question. Only those which are `[JSImport]` with return type `Task` make sense.
I reviewed all of them now
```
ws_wasm_close
ws_wasm_receive
ws_wasm_send
ws_wasm_open

http_wasm_get_response_length
http_wasm_get_streamed_response_bytes
http_wasm_fetch_bytes
http_wasm_fetch
```